homeassistant/components/epic_games_store/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for Epic Games Store integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from epicstore_api import EpicGamesStoreAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.selector import LanguageSelector, LanguageSelectorConfig
+
+from .const import CONF_LANGUAGE, CONF_SUPPORTED_LANGUAGES, DOMAIN
+from .helper import get_country_from_language
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_LANGUAGE): LanguageSelector(
+            LanguageSelectorConfig(languages=CONF_SUPPORTED_LANGUAGES)
+        ),
+    }
+)
+
+
+def get_default_language(hass: HomeAssistant) -> str:
+    """Get default language code based on Home Assistant config."""
+    language_code = f"{hass.config.language}-{hass.config.country}"
+    if language_code in CONF_SUPPORTED_LANGUAGES:
+        return language_code
+    if hass.config.language in CONF_SUPPORTED_LANGUAGES:
+        return hass.config.language
+    return "en-US"
+
+
+async def validate_input(hass: HomeAssistant, user_input: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect."""
+    api = EpicGamesStoreAPI(
+        user_input[CONF_LANGUAGE], get_country_from_language(user_input[CONF_LANGUAGE])
+    )
+    data = await hass.async_add_executor_job(api.get_free_games)
+
+    if data.get("errors"):
+        _LOGGER.warning(data["errors"])
+
+    try:
+        data["data"]["Catalog"]["searchStore"]["elements"]
+    except Exception:  # pylint: disable=broad-except
```

Comment:
I think what we are ultimately catching is a KeyError right?

Reply:
`TypeError` or `KeyError`, more a Type as on test `tests/components/epic_games_store/test_config_flow.py::test_form_cannot_connect_wrong_param`

See

```shell
2023-12-28 11:01:23.456 WARNING  MainThread homeassistant.components.epic_games_store.config_flow:config_flow.py:48 [{'message': 'CatalogQuery/searchStore: Request failed with status code 400', 'locations': [{'line': 18, 'column': 9}], 'correlationId': 'e10ad58e-a4f9-4097-af5d-cafdbe0d8bbd', 'serviceResponse': '{"errorCode":"errors.com.epicgames.catalog.invalid_country_code","errorMessage":"Sorry the value you entered: en-US, does not appear to be a valid ISO country code.","messageVars":["en-US"],"numericErrorCode":5222,"originatingService":"com.epicgames.catalog.public","intent":"prod","errorStatus":400}', 'stack': None, 'path': ['Catalog', 'searchStore']}]
2023-12-28 11:01:23.456 ERROR    MainThread homeassistant.components.epic_games_store.config_flow:config_flow.py:82 Unexpected exception
Traceback (most recent call last):
  File "/workspaces/home-assistant-core/homeassistant/components/epic_games_store/config_flow.py", line 78, in async_step_user
    await validate_input(self.hass, user_input)
  File "/workspaces/home-assistant-core/homeassistant/components/epic_games_store/config_flow.py", line 51, in validate_input
    data["data"]["Catalog"]["searchStore"]["elements"]
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable
```

But this case should not happen now as of new version of lib, so I refactored it in c0a3429e6063c38f7f67d0d56159df7f5077b2fd

Reply:
fixed in 14ca00a3c56cea41965ac676943957d035a86f37

Reply:
Apply fix again in 5eda545a4695ad9956abe8a89e88b5e5ecb05200